internal/services/network/subnet_resource.go

```diff
@@ -320,14 +392,36 @@ func resourceSubnetUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 		props.Delegations = expandSubnetDelegation(delegationsRaw)
 	}
 
-	if d.HasChange("enforce_private_link_endpoint_network_policies") {
-		v := d.Get("enforce_private_link_endpoint_network_policies").(bool)
-		props.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(v))
-	}
+	if features.FourPointOhBeta() {
+		if d.HasChange("private_endpoint_network_policies_enabled") {
+			v := d.Get("private_endpoint_network_policies_enabled").(bool)
+			props.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetNetworkPolicy(v))
+		}
 
-	if d.HasChange("enforce_private_link_service_network_policies") {
-		v := d.Get("enforce_private_link_service_network_policies").(bool)
-		props.PrivateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(v))
+		if d.HasChange("private_link_service_network_policies_enabled") {
+			v := d.Get("private_link_service_network_policies_enabled").(bool)
+			props.PrivateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetNetworkPolicy(v))
+		}
+	} else {
+		if d.HasChange("enforce_private_link_endpoint_network_policies") || d.HasChange("private_endpoint_network_policies_enabled") {
+			if d.HasChange("enforce_private_link_endpoint_network_policies") {
+				v := d.Get("enforce_private_link_endpoint_network_policies").(bool)
+				props.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandEnforceSubnetNetworkPolicy(v))
+			} else {
+				v := d.Get("private_endpoint_network_policies_enabled").(bool)
+				props.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetNetworkPolicy(v))
+			}
+		}
+
+		if d.HasChange("enforce_private_link_service_network_policies") || d.HasChange("private_link_service_network_policies_enabled") {
+			if d.HasChange("enforce_private_link_service_network_policies") {
```

Comment:
As above, if `enforce_private_link_service_network_policies` is unset, the new value in `private_link_service_network_policies_enabled` won't be read.

Reply:
These fields are mutually exclusive from each other, since in the schema they conflict with one another... so I believe what will happen here is:

* Both fields are removed from the config, it will just persist whatever the values are at that point in time since these are both now computed optional.
* If one is changed the other cannot be changed so it will update the value in azure to match what was set in the config, then in the read the value of the other field will be corrected to mirror the value that was set in the update CRUD operation.

At least that is what I am thinking, am I overlooking something? 🤔

Reply:
**NOTE:** I have also added two new test cases to validate that the fields can override each other as expected:
* `TestAccSubnet_PrivateLinkPoliciesToggleWithEnforceFirst`
* `TestAccSubnet_PrivateLinkPoliciesToggleWithEnabledFirst`

I am going to use the age old dev mantra.... Works on my box... lol... 🤣